components/board.tsx

```diff
@@ -58,6 +54,10 @@ export default function Board(props: Props) {
         ...state.next,
         played: { correct },
       });
+      
+      const newNext = state.nextButOne;
+      const newNextButOne = getRandomItem(newDeck, newNext ? [...newPlayed, newNext] : newPlayed);
+      const newImageCache = [preloadImage(newNextButOne.image)];
```

Comment:
Just a location change.

Reply:
I did the relocation because of the line 'newPlayed.splice' which modifies the array.